include/uk/plat/lcpu.h

```diff
@@ -41,6 +41,20 @@
 extern "C" {
 #endif
 
+#define UKPLAT_PER_LCPU_DEFINE(var_type, var_name) \
+	var_type var_name[CONFIG_UKPLAT_LCPU_MAXCOUNT]
+#define ukplat_per_lcpu(var_name, lcpu_idx) \
+	var_name[lcpu_idx]
+#define ukplat_per_lcpu_current(var_name) \
+	ukplat_per_lcpu(var_name, ukplat_lcpu_idx())
+
+#define UKPLAT_PER_LCPU_ARRAY_DEFINE(var_type, var_name, size) \
+	var_type var_name[CONFIG_UKPLAT_LCPU_MAXCOUNT][size]
+#define ukplat_per_lcpu_array(var_name, lcpu_idx, idx) \
+	var_name[lcpu_idx][idx]
+#define ukplat_per_lcpu_array_current(var_name, idx) \
+	ukplat_per_lcpu_array(var_name, ukplat_lcpu_idx(), idx)
+
```

Comment:
Hey @andraprs,
I would move this down here, because it practically depends on `ukplat_lcpu_idx()` and also feels a bit out of context right at the top:
```C
#else /* CONFIG_HAVE_SMP */
#define ukplat_lcpu_idx()	(0)
#define ukplat_lcpu_count()	(1)
#endif /* CONFIG_HAVE_SMP */

/* Per-LCPU variable definition */
<----------

#ifdef __cplusplus
}
#endif
```

Other than that I think this is fine 😃

Reply:
Done. Thank you for review.